### PR TITLE
Sorting function

### PR DESCRIPTION
**Pull request by SeR4h**
@dennisja @billkabanga Please review my sorting algorithm

### CODE REVIEW

File: arraySort.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+    if (x[i] % 2 == 0) {
+        evenNumbers.push(x[i]);
+    }
+    if (x[i] % 2 == 1) {
+        oddNumbers.push(x[i]);
+    }
+    i++;
+}
+console.log(props)
```
Comment (dennisja) [inline arraySort.js]: Why are you logging props here, Where do you get the props variable from?

File: arraySort.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+function arraySort(a, b) {
+    return b - a
+    //return a<b?1 : a>b?-1 : 0
+}
+
+var x = [1, 4, 7, "x", 3, 5, 8, "t", 0, 6, 10, "a"];
+x.sort(arraySort)
+x.reverse()
+i = 0
```
Comment (dennisja) [inline arraySort.js]: We talked about this previously, Always avoid using variables before declaring them. This is a bad practice in JS it adds them to the global scope

File: arraySort.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+function arraySort(a, b) {
+    return b - a
+    //return a<b?1 : a>b?-1 : 0
+}
+
+var x = [1, 4, 7, "x", 3, 5, 8, "t", 0, 6, 10, "a"];
+x.sort(arraySort)
```
Comment (dennisja) [inline arraySort.js]: Why are you even sorting this array?

File: arraySort.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+x.reverse()
```
Comment (dennisja) [inline arraySort.js]: Why are you even reversing this?

File: arraySort.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+i = 0
+while (i < x.length) {
+    if (isNaN(x[i])) {
+        characters.push(x[i]);
+    }
+    if (x[i] % 2 == 0) {
+        evenNumbers.push(x[i]);
+    }
+    if (x[i] % 2 == 1) {
+        oddNumbers.push(x[i]);
```
Comment (dennisja) [inline arraySort.js]: `oddNumbers` is not defined

File: arraySort.js (hunk 1 above)
Comment (SeR4h) [reply arraySort.js]: @dennisja my object was mistakenly cut before i commited. The props is supposed to be up. I've added the actual code on the second commit

File: arraySort.js (hunk 4 above)
Comment (SeR4h) [reply arraySort.js]: @dennisja i did to match with what the question wanted

File: arraySort.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+function arraySort(a, b) {
```
[the remark this answers is not in the document]
Comment (SeR4h) [reply arraySort.js]: @dennisja see question below
Given an array containing numbers and one character strings, write a function that  takes the array and returns an object with property names evens, odds, and chars.  The value for evens is an array of sorted even numbers and odds is an array of  sorted odd numbers and chars is an array of sorted character strings.   
arraySort([2,0,6,5,1,7,’z’,’a’])  
# returns {‘evens’: [0,2,6], odds: [1,5,7], chars:[‘a’,’z’]}

File: arraySort.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    //return a<b?1 : a>b?-1 : 0
+}
+
+var x = [1, 4, 7, "x", 3, 5, 8, "t", 0, 6, 10, "a"];
+x.sort(arraySort)
+x.reverse()
+i = 0
+while (i < x.length) {
+    if (isNaN(x[i])) {
+        characters.push(x[i]);
```
[the remark this answers is not in the document]
Comment (SeR4h) [reply arraySort.js]: please review modified sorting function

File: arraySort.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+var evenNumbers= new Array();
```
Comment (dennisja) [inline arraySort.js]: - You were told to create a function that takes an array and returns an object. Can you do it as you were requested, please?

**Pull request merged**